ArjunaJTS/integration/src/main/java/com/arjuna/ats/jbossatx/BaseTransactionManagerDelegate.java

```diff
@@ -336,6 +336,11 @@ public void unlock(final TransactionLocal local, final Transaction transaction)
     // Bad things will probably happen if users ever use this key themselves
     private final String LOCKS_MAP = "__LOCKS_MAP";
 
+    // Using string LOCKS_MAP directly as object of synchronization is not recommended
+    // String literals are centrally interned and could also be locked on by a library,
+    // potentially having deadlocks or lock collisions
+    private final Object locksMapLock = new Object();
```

Comment:
Hi @tomjenkinson , I would like check with you my doubt of this change. I'm not sure if it could be that straightforward as I suggest here. Could not happen `locks` being filled with an empty map from two threads when `locks` variable is empty? It seems that synchronization block starts after `locks` is searched and after is checked to be null. Thanks.